--- ../src-base/minecraft/net/minecraft/entity/item/EntityMinecartTNT.java
+++ ../src-work/minecraft/net/minecraft/entity/item/EntityMinecartTNT.java
@@ -1,5 +1,6 @@
 package net.minecraft.entity.item;
 
+import carpet.CarpetSettings;
 import net.minecraft.block.BlockRailBase;
 import net.minecraft.block.state.IBlockState;
 import net.minecraft.entity.Entity;
@@ -98,7 +99,10 @@
 
             if (!p_94095_1_.func_94541_c() && this.field_70170_p.func_82736_K().func_82766_b("doEntityDrops"))
             {
-                this.func_70099_a(new ItemStack(Blocks.field_150335_W, 1), 0.0F);
+                if (!CarpetSettings.minecartFullDropBackport)
+                {
+                    this.func_70099_a(new ItemStack(Blocks.field_150335_W, 1), 0.0F);
+                }
             }
         }
         else
@@ -122,7 +126,8 @@
                 d0 = 5.0D;
             }
 
-            this.field_70170_p.func_72876_a(this, this.field_70165_t, this.field_70163_u, this.field_70161_v, (float)(4.0D + this.field_70146_Z.nextDouble() * 1.5D * d0), true);
+            this.field_70170_p.func_72876_a(this, this.field_70165_t, this.field_70163_u, this.field_70161_v,
+                    (float)(4.0D + this.field_70146_Z.nextDouble() * 1.5D * d0), true);
             this.func_70106_y();
         }
     }
